resonantLimits/write_res_card.py

```diff
@@ -14,539 +14,453 @@ def parseOptions():
+    svdir = select + '_' + variable
+    dname = opt.outdir + '_' + opt.channel
+    out_dir = os.path.join(opt.indir, 'cards_{}/{}/'.format(dname, svdir))
     cmd = 'mkdir -p {}'.format(out_dir)
     os.system(cmd)
-        
-    regionName = ["","regB","regC","regD"]
-    regionSuffix = ["SR","SStight","OSinviso","SSinviso"]
-    thechannel = "2"
-    if opt.channel == "ETau" : thechannel="1"
-    elif opt.channel == "MuTau" : thechannel = "0"
 
     #read config
     categories = []
     categories.append((0,select))
     MCbackgrounds=[]
     processes=[]
     inRoot = ROOT.TFile.Open(opt.filename)
+
     for bkg in backgrounds:
-        #Add protection against empty processes => If I remove this I could build all bins at once instead of looping on the selections
-        templateName = "{0}_{1}_SR_{2}".format(bkg,select,variable[theCat])
+        templateName = '{}_{}_SR_{}'.format(bkg, select, variable)
         try:
             template = inRoot.Get(templateName)
-        except ReferenceError:
-            print('File: {}'.format(opt.filename))
+            if template.Integral() > 1E-6:
+                processes.append(bkg)
+                if bkg != 'QCD':
+                    MCbackgrounds.append(bkg)
+                    
+        except (ReferenceError, AttributeError):
+            print('\nFile: {}'.format(opt.filename))
             print('Template: {}'.format(templateName))
             raise
-        template.Print()
-        if template.Integral()>0.000001:
-            processes.append(bkg)
-            if bkg != "QCD":
-                MCbackgrounds.append(bkg)
 
     rates = []
     iQCD = -1
     totRate = 0
-    templateName = "data_obs_{0}_{2}_{1}".format(select,variable[theCat],regionSuffix[region])
+    suffix_str = '{}_{}_{}'.format(select, regions[0], variable)
+    templateName = 'data_obs_' + suffix_str
     template = inRoot.Get(templateName)
     obs = template.GetEntries()
 
     for proc in range(len(backgrounds)):
-        if "QCD" in backgrounds[proc] and region > 0:
+        if 'QCD' in backgrounds[proc] and regions[0] != 'SR':
             rates.append(-1)
             iQCD = proc
-        else :
-            templateName = "{0}_{1}_{3}_{2}".format(backgrounds[proc],select,variable[theCat],regionSuffix[region])
+        else:
+            templateName = '{}_'.format(backgrounds[proc]) + suffix_str
             template = inRoot.Get(templateName)
             brate = template.Integral()
             rates.append(brate)
             totRate = totRate + brate
-    if iQCD >= 0 and region > 0: rates[iQCD] = ROOT.TMath.Max(0.0000001,obs-totRate)
-    #print(str(region), rates)
+    if iQCD >= 0 and regions[0] != 'SR':
+        rates[iQCD] = ROOT.TMath.Max(0.0000001,obs-totRate)
     
-    for proc in range(len(signals)):
-        templateName = '{0}_{1}_{3}_{2}'.format(signals[proc],select,variable[theCat],regionSuffix[0])
-        template = inRoot.Get(templateName)
-        if template.Integral() <= 0:
```

Comment:
In my humble opinion, for reference, this is the line causing issues. If we are looking at the mass category with mHH > 1100 GeV it is quite unlikely that the histogram for the 250 GeV signals has any entry so the corresponding datacards are not written out. Once we move to the workspace creation combine is looking for datacards for each mass point in parallel and this raises and exception at runtime.